src/statmodels.jl

```diff
@@ -460,6 +460,14 @@ struct PValue
 end
 PValue(p::PValue) = p
 
+float(p::PValue) = float(p.v)
+
+for op in [Symbol("=="), :≈, :<, :≤, :>, :≥]
```

Comment:
```suggestion
for op in [:(==), :≈, :<, :≤, :>, :≥]
```
Maybe also define `isequal` and `isless`?

Actually I wonder whether it would be simpler to define `promote_type(::Type{PValue{S}}, y::T) where {S, T<:Real} = promote_type(S, T)`, making `S` a type parameter with `v::S`. Then I think we'll get all operations on numbers defined for free.

Reply:
You get `isequal` and `isless` for free with the symbolic forms (and I use `isequal` and `isless`) in the tests.

I was also thinking about whether `promote_type` might be the easier route. I'll try that later. :)

Reply:
Ah yes so `<` falls back to `isless`, but it's not correct for `NaN`, right?

Reply:
Uh.... good question

Reply:
Oh wait, I know why I didn't do `promote_type`: that would be technically breaking, both for `PValue` and `TestStat`. We're pre 1.0 so another release isn't a big deal. But that might be something to get one more opinion on (e.g. @ararslan).

(I also want `PValue` and `TestStat` to have a consistent interface between themselves since they're both essentially just pretty printers for a semantically annotated `Real`.)

Reply:
You mean it would be breaking if you do something like `[pval, 1.0]`? Yes that would be slightly breaking so that would probably warrant tagging a minor release.

Reply:
Yeah, and adding a type parameter to a struct also breaks (de)serialization.

Reply:
Maybe let's keep your current approach then, and only use `promote_type` in another PR that would be merged only once we have other breaking changes to tag.

Reply:
As soon as this one (and CoefTable show method) lands, I'll open that PR :smile